core/src/test/java/org/apache/calcite/test/JdbcTest.java

```diff
@@ -5898,17 +5965,32 @@ private void checkMapSchemaDirectConnection(String s) throws SQLException {
             + "order by \"empid\" limit 2", null);
     with
         .query("select \"name\" from \"adhoc\".V order by \"name\"")
+        .withHook(Hook.SQL2REL_CONVERTER_CONFIG_BUILDER,
+            (Consumer<Holder<Config>>) configHolder ->
+                configHolder.set(configHolder.get().withRemoveSortInSubQuery(false)))
         .returns("name=Bill\n"
             + "name=Theodore\n");
 
     // Now a sub-query with ORDER BY and LIMIT clauses. (Same net effect, but
     // ORDER BY and LIMIT in sub-query were not standard SQL until SQL:2008.)
+    with
+        .query("select \"name\" from (\n"
+            + "select * from \"adhoc\".\"EMPLOYEES\" where \"deptno\" = 10\n"
+            + "order by \"empid\" limit 2)\n"
+            + "order by \"name\"")
+        .withHook(Hook.SQL2REL_CONVERTER_CONFIG_BUILDER,
+            (Consumer<Holder<SqlToRelConverter.Config>>) configHolder ->
+                configHolder.set(configHolder.get().withRemoveSortInSubQuery(false)))
+        .returns("name=Bill\n"
+            + "name=Theodore\n");
+
     with
         .query("select \"name\" from (\n"
             + "select * from \"adhoc\".\"EMPLOYEES\" where \"deptno\" = 10\n"
             + "order by \"empid\" limit 2)\n"
             + "order by \"name\"")
         .returns("name=Bill\n"
+            + "name=Sebastian\n"
             + "name=Theodore\n");
```

Comment:
This is wrong. The LIMIT should not be removed. The sub-query should continue to return two rows.

Reply:
You are right, I modified the code to preserve the previous behavior.